templates/template_index.html

```diff
@@ -206,18 +206,78 @@ <h2 class="mb-4 text-center">Compatibility</h2>
   }
 }
 
+var titlefilter = null;
+var statusfilter = null;
+var hiddenstatuses = []; 
+
 // Simple filter via: https://stackoverflow.com/a/44736154
 $("#filter").keyup(delay(function() {
-  var filter = $(this).val();
+  titlefilter = $(this).val();
+  filtertitles();
+}, 250));
+
+function filterstatus(status)
+{
+  if(statusfilter != null && statusfilter === status)
+  {
+    //User has toggled same status again, show all titles
+    statusfilter = null;
+  }
+  else{
+    statusfilter = status;
+  }
+
+  filtertitles();
+}
+
+function hidestatus(statustohide)
+{
+  if(hiddenstatuses.includes(statustohide))
+  {
+    //User has toggled a hidden status, remove status from hidden filter
+    hiddenstatuses = hiddenstatuses.filter(function(hiddenstatus) {
+    return hiddenstatus !== statustohide;
+    });
+  }
+  else{
+    hiddenstatuses.push(statustohide);
+  }
+
+  filtertitles();
+}
+
+function filtertitles()
+{
   $('.title-card').each(function() {
-    if ($(this).attr('data-title-name').search(new RegExp(filter, "i")) < 0) {
-      $(this).hide();
-    } else {
+    var shouldshow = true;
+
+    //Title name matches input
+    if(titlefilter != null && $(this).attr('data-title-name').search(new RegExp(titlefilter, "i")) < 0 )
```

Comment:
Let's restructure this a bit to be a little faster. Right now this code checks all three conditions, every time, for every title. We can write it to short-circuit when any condition matches, starting with faster/broader exclusion criteria, and only doing more complex testing (regex) if needed. e.g.

```javascript
if ((<check status filter>) ||
    (<check hidden filter>) ||
    (<regex title test>)) {
  $(this).hide();
} else {
  $(this).show();
}
```

You can use if/else if if you prefer, or use the boolean expression like I did above, your call.